### ISSUE TITLE
BubbleUPNP devices not being discovered

### ISSUE THREAD

**Issue by kenkuo** [dheijl/swyh-rs#157]
[swyh.txt](https://github.com/user-attachments/files/17874356/swyh.txt)

The devices are responding to SSDP but not sure what's going on after that. These all show up in swyh original but not in -rs

**Comment by dheijl** [comment]
please attach the complete (unedited) debug log.

**Comment by kenkuo** [comment]
[log.txt](https://github.com/user-attachments/files/17875709/log.txt)

**Comment by dheijl** [comment]
Thanks.

The problem is simple: the location header in the SSDP response *should* be uppercase "LOCATION:", but for some reason Bubble uses "Location:". You will see in your log that all other devices properly use LOCATION.

Easily fixed, probably tomorrow.

**Comment by dheijl** [comment]
Should be fixed in [release 1.12.1](https://github.com/dheijl/swyh-rs/releases/tag/1.12.1).

**Comment by dheijl** [comment]
The reason: it's the only device I have ever encountered that puts the Location header last.
Everyone else puts it at the top, and I stopped scanning at the St header, so the Location header was missed.
I'll yank 1.12.1 and release 1.12.2.

**Comment by dheijl** [comment]
[This time](https://github.com/dheijl/swyh-rs/releases/tag/1.12.2) I hope I fixed it properly. Sorry for the mess...

**Comment by kenkuo** [comment]
[log.txt](https://github.com/user-attachments/files/17906706/log.txt)
Almost there :)
Now only "Living room speaker" shows up in UI but I can see all the other devices are being discovered now in the log. 
There should be 4 devices exposed from BubbleUPNP as DLNA:
Living room speaker
Master Bedroom speaker
Entryway speaker
Speaker group

Let me know if you need any more info from me, thanks for your quick responses

**Comment by kenkuo** [comment]
If I expose an openhome renderer in BubbleUPNP that one takes the place of the "Living room speaker (DLNA)" renderer in the UI. Possibly only the first BubbleUPNP renderer is being properly shown?
[log.txt](https://github.com/user-attachments/files/17906858/log.txt)

**Comment by dheijl** [comment]
If I release a new beta to test, do you need the setup or is a debug binary sufficient?

**Comment by kenkuo** [comment]
If it's just replacing the swyh-rs.exe I can handle that

**Comment by dheijl** [comment]
There's a [new beta](https://github.com/dheijl/swyh-rs/releases/tag/1.12.3-beta) setup.

**Comment by dheijl** [comment]
I expected problems but I thought the buttons for the players would at least show up.

However, since the streaming is totally disconnected from UPNP, I have no means to connect a device that is currently streaming to the buttons in the GUI, and I don't think I can solve that. I only have the IP address of any device that starts streaming, and nothing more. So it's probably not something that I can support reliably.

**Comment by dheijl** [comment]
But I'll see if I can up with something that works if you are prepared to wait a bit.

**Comment by dheijl** [comment]
The beta has been [updated](https://github.com/dheijl/swyh-rs/releases/tag/1.12.3-beta). I think it should work now.

Some things will probably no longer work, let me now if you find something (or if it still doesn't work at all).

**Comment by kenkuo** [comment]
Looks to be the same with beta 2, albeit with no regression either from what I can tell.
[log.txt](https://github.com/user-attachments/files/17926391/log.txt)

**Comment by dheijl** [comment]
The beta has been [updated](https://github.com/dheijl/swyh-rs/releases/tag/1.12.3-beta) again. I think it should work now, honestly :)

Some things will probably no longer work, let me now if you find something (or if it still doesn't work at all).

**Comment by kenkuo** [comment]
Beta 3 also looks to be exact same behavior, still able to stream to the one exposed dlna device just fine:
[log.txt](https://github.com/user-attachments/files/17938489/log.txt)

**Comment by dheijl** [comment]
For some reason you're still running the original beta. This line:
```
Renderer Denon AVR-S760H Denon AVR-S760H ip 192.168.1.243 at urlbase http://192.168.1.243:60006/ has 9 services
```
should be similar to this one:
```
Renderer Marantz UPNP UpMPD ip 192.168.0.131 at location http://192.168.0.131:49152/uuid-fdfabe96-b6c6-92ff-31dd-dca63239e1f0/description.xml has 7 services
```
I have no idea how this is possible, as I delete the old setup every time and replace it with the new one on Github.

But it certainly explains why you don't see any changes...

**Comment by dheijl** [comment]
I downloaded the setup from Github and it installed the latest version, showing the correct messages in the log.

I replaced it again, but this time the app should show the version 1.12.3-b3 in the title bar:

![image](https://github.com/user-attachments/assets/78f7d4ff-70b7-4a09-8145-a6443d5b09c1)

**Comment by dheijl** [comment]
I have no idea if Bubble or Chromecast support changing the volume, reading it out seems to work though.

Looking at your screenshot, I see that the SSDP interval is extremely short, if you're on shaky WiFi connections it might interfere with the streaming.

I will wait a couple of days before releasing this officially, in case you find any problems.

One restriction when using devices sharing the same IP address: if you start streaming from swyh-rs, and terminte it with some other means (breaking the streaming connection from the device end), swyh-rs is not able to reflect this on the button, it will remain lighted even though the device is no longer streaming.

**Comment by dheijl** [comment]
Oh, and thanks for testing this and bearing with my attempts to fix it. Much appreciated!

**Comment by dheijl** [comment]
Fixed with release 1.12.3.
